src/theme/mod.rs

```diff
@@ -61,75 +110,130 @@ impl Theme {
+        .map(|dir_entry| dir_entry.path())
+        .find(|p| looks_like_js(p, js_library))
+}
 
-        // highlight.js
-        if let Ok(mut f) = File::open(&src.join("highlight.js")) {
-            theme.highlight_js.clear();
-            let _ = f.read_to_end(&mut theme.highlight_js);
-        }
+/// Does this file look like a valid `jquery.js`?
+fn looks_like_js(path: &PathBuf, library_name: &str) -> bool {
+    let filename = match path.file_name().and_then(|os_str| os_str.to_str()) {
+        Some(ext) => ext,
+        None => return false,
+    };
 
-        // clipboard.js
-        if let Ok(mut f) = File::open(&src.join("clipboard.min.js")) {
-            theme.clipboard_js.clear();
-            let _ = f.read_to_end(&mut theme.clipboard_js);
-        }
+    let pattern = format!(r"^{}(-\d(\.\d)*)?(\.min)?\.js", library_name);
 
-        // store.js
-        if let Ok(mut f) = File::open(&src.join("store.js")) {
-            theme.store_js.clear();
-            let _ = f.read_to_end(&mut theme.store_js);
-        }
+    Regex::new(&pattern).expect("Invalid regex").is_match(
+        &filename,
+    )
+}
 
-        // highlight.css
-        if let Ok(mut f) = File::open(&src.join("highlight.css")) {
-            theme.highlight_css.clear();
-            let _ = f.read_to_end(&mut theme.highlight_css);
-        }
 
-        // tomorrow-night.css
-        if let Ok(mut f) = File::open(&src.join("tomorrow-night.css")) {
-            theme.tomorrow_night_css.clear();
-            let _ = f.read_to_end(&mut theme.tomorrow_night_css);
-        }
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use tempdir::TempDir;
+    use std::path::PathBuf;
+
+    #[test]
+    fn theme_uses_defaults_with_nonexistent_src_dir() {
+        let non_existent = PathBuf::from("/non/existent/directory/");
+        assert!(!non_existent.exists());
+
+        let should_be = Theme::default();
+        let got = Theme::new(&non_existent);
+
+        assert_eq!(got, should_be);
+    }
 
-        // ayu-highlight.css
-        if let Ok(mut f) = File::open(&src.join("ayu-highlight.css")) {
-            theme.ayu_highlight_css.clear();
-            let _ = f.read_to_end(&mut theme.ayu_highlight_css);
+    #[test]
+    fn theme_dir_overrides_defaults() {
+        // Get all the non-Rust files in the theme directory
+        let special_files = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+            .join("src/theme")
+            .read_dir()
+            .unwrap()
+            .filter_map(|f| f.ok())
+            .map(|f| f.path())
+            .filter(|p| p.is_file() && !p.ends_with(".rs"));
+
+        let temp = TempDir::new("mdbook").unwrap();
+
+        // "touch" all of the special files so we have empty copies
+        for special_file in special_files {
+            let filename = temp.path().join(special_file.file_name().unwrap());
+            let _ = File::create(&filename);
         }
 
-        theme
+        let got = Theme::new(temp.path());
```

Comment:
How about?

```rust
        let empty = Theme {
            index: vec![],
            css: vec![],
            favicon: vec![],
            js: vec![],
            highlight_css: vec![],
            tomorrow_night_css: vec![],
            ayu_highlight_css: vec![],
            highlight_js: vec![],
            clipboard_js: vec![],
            store_js: vec![],
            jquery: vec![],
        };

        let got = Theme::new(temp.path());

        assert_eq!(got, empty);
```

This might be subjectively a little less repetitive and forces us to actually keep the test in sync with `Theme` struct impl which otherwise be hard